diagnostics/eulerian_storm_track/eulerian_storm_track_util.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+############# EULERIAN STROM TRACKER ############
+###### Created by: Jeyavinoth Jeyaratnam     ####
+###### Created Date: 03/29/2019              ####
+###### Last Modified: 05/09/2019             ####
+#################################################
+
+# Importing standard libraries
+import numpy as np 
+import datetime as dt
+import os
+from netCDF4 import Dataset
+import warnings
+
+def transient_eddies(daily_data):
+  '''
+  Data has to be provided as daily_data
+  it will compute the difference between the current day and the previous day
+  i.e. X(t+1) - X(t), nans for the last index 
+  in Booth et al., 2017, vprime = (x(t+1) - x(t))/2. 
+  '''
+  # pad the right of the array with nan values along the first dimension
+  # then extract the values from the 2nd column (index = 1) to the end
+  # this will give us a shifted array of daily data, i.e. X(t+1)
+  daily_data_shift = np.pad(daily_data, ((0,1), (0,0), (0,0)), mode='constant', constant_values=np.nan)[1:, :, :]
+
+  return (daily_data_shift - daily_data)/2. # X(t+1) - X(t), with nan values for the last time dimension
+
+def model_std_dev(data, start_year, time, season='djf'):
+  '''
+  Data input should be in the format (time, lat, lon)
+  We will calculate the std_dev for the given time_ind, the time_ind has to be a logical array of the size of the time dimension 
+  '''
+  # convert time as numpy array
+  time = np.asarray(time)
+
+  # getting the datetime values for the time index
+  dates_month=[]
+  dates_year=[]
+  for i_time in time: 
+    temp_time = dt.datetime(start_year, 1, 1) + dt.timedelta(days=np.float(i_time)-1)
+    dates_month.append(temp_time.month)
+    dates_year.append(temp_time.year)
+
+  dates_month = np.asarray(dates_month)
+  dates_year = np.asarray(dates_year)
+
+  eddy_year = []
+  for i_year in range(int(os.environ['FIRSTYR']), int(os.environ['LASTYR'])+1):
+    if (season == 'djf'):
+      time_ind = ((dates_year == i_year) & (dates_month == 1)) | ((dates_year == i_year) & (dates_month == 2)) | ((dates_year == i_year-1) & (dates_month == 12)) 
+    elif (season == 'mam'):
+      time_ind = ((dates_year == i_year) & (dates_month == 3)) | ((dates_year == i_year) & (dates_month == 4)) | ((dates_year == i_year) & (dates_month == 5)) 
+    elif (season == 'jja'):
+      time_ind = ((dates_year == i_year) & (dates_month == 6)) | ((dates_year == i_year) & (dates_month == 7)) | ((dates_year == i_year) & (dates_month == 8)) 
+    elif (season == 'son'):
+      time_ind = ((dates_year == i_year) & (dates_month == 9)) | ((dates_year == i_year) & (dates_month == 10)) | ((dates_year == i_year) & (dates_month == 11)) 
+
+    with warnings.catch_warnings():
+      warnings.simplefilter("ignore", category=RuntimeWarning)
```

Comment:
Is this to catch warnings issued by operating on arrays that contain NaNs?

Reply:
Yes it is.